client/blocks/comments/post-comment-list.jsx

```diff
@@ -210,6 +232,30 @@ class PostCommentList extends React.Component {
 							} )
 						}</span> : null }
 				</div> }
+				{ isEnabled( 'comments/filters-in-posts' ) &&
+					<SegmentedControl compact primary>
+						<SegmentedControlItem
+							selected={ commentsFilter === 'all' }
+							onClick={ this.handleFilterClick( 'all' ) }>{ translate( 'All' ) }
```

Comment:
this creates a new handler function every time render is called, which causes the SegementedControlItem to rerender too. It would be better to create the action specific handlers once, as props, and then use those bound handlers. So up on the class def:

```
handleAllFilterClick = () => this.props.onFilterChange( 'all' );
// or
handleAllFilterClick = this.createFilterClickHandler( 'all' ); // renamed handleFilterClick
```